src/esp/core/ManagedContainer.h

```diff
@@ -336,7 +339,11 @@ class ManagedContainer : public ManagedContainerBase {
       return nullptr;
     }
     auto orig = getObjectInternal<T>(objectHandle);
-    return this->copyObject(orig);
+    if (AccessViaCopies) {
```

Comment:
`if constexpr`? (does habitat use c++17 or not?)

Same below.

Reply:
We're not c++17 yet, unfortunately.